src/app/core/services/electron/electron.service.ts

```diff
@@ -66,6 +67,16 @@ export class ElectronService {
     return result;
   }
 
+  async addPartToDatabase(updatedParts: UpdatedPart[]) {
+    const partsToUpdate: UpdatedPart[] = [];
+
+    for (const part of updatedParts) {
```

Comment:
Use filter operator on Array. In general you are using way too much for loops than its necessary. Embrace built in functions. But also bear in mind consequences as well. Each array pass increases a big O complexity. If you can do more than one thing in a single pass use for loop otherwise there is no point.